Connection Type Translator Comments

jarolrod opened the pull request:
This PR introduces Qt translator comments for `Connection Type` strings in `guiutil.cpp` as well as `rpcconsole.cpp`.

This is an alternate implementation of the idea presented in the last three commits of #289. It is especially inspired by commit 842f4e834dfe5fd2786a5092f78ea28da1b36e4f.

Per [Qt Dev Notes](https://github.com/bitcoin-core/bitcoin-devwiki/wiki/Developer-Notes-for-Qt-Code), it is better to not break up strings when not necessary. This way we preserve the full context for translators.

jarolrod commented:
updated from 0819d50 -> f5c2377 ([pr345.01](https://github.com/jarolrod/gui/tree/pr345.01) -> [pr345.02](https://github.com/jarolrod/gui/tree/pr345.02), [diff](https://github.com/jarolrod/gui/compare/pr345.01...jarolrod:pr345.02))

**Changes:**
- Add translator comments for some missed connection type strings.

hebasto commented on src/qt/rpcconsole.cpp:
```diff
@@ -471,14 +471,28 @@ RPCConsole::RPCConsole(interfaces::Node& node, const PlatformStyle *_platformSty
+            Outbound connections. */
         tr("Outbound Full Relay: default"),
+        /*: Explanatory text for an Outbound peer connection which
+            only relays network information about Blocks. */
```
For translators it is not obvious the information relayed through the network is related to blocks, transactions, addresses, and service needs (handshaking, ping etc). _"only relays network information about Blocks"_ is not the same as _"does not relay transactions or addresses"_ to them. Btw, the latter seems more correct as `BLOCK_RELAY` peers handle service network commands well.

jonatack replied:
Yes. 

(It's not clear to me what value these descriptions add, as they are next to descriptions that already exist.)

hebasto replied:
> It's not clear to me what value these descriptions add, as they are next to descriptions that already exist.

If understand your concern correctly, the point is that in Transifex.com on-line editor a translator can see only a translatable string and an attached comment, without surrounding code.

jonatack replied:
Thanks for the explanation. That makes sense.

hebasto reviewed (commented):
Approach ACK f5c23778e54b74cdd4f8fe8afd507d0a0e045d37.

jonatack commented on src/qt/rpcconsole.cpp:
```diff
@@ -471,14 +471,28 @@ RPCConsole::RPCConsole(interfaces::Node& node, const PlatformStyle *_platformSty
+            is used to test the validity of known addresses. */
         tr("Outbound Feeler: short-lived, for testing addresses"),
+        /*: Explanatory text for an Outbound peer connection which is used
+            to get addresses from a peer. */
```
s/get/request/

jonatack commented on src/qt/rpcconsole.cpp:
```diff
@@ -471,14 +471,28 @@ RPCConsole::RPCConsole(interfaces::Node& node, const PlatformStyle *_platformSty
     const std::vector<QString> CONNECTION_TYPE_DOC{
+        //: Explanatory text for an Inbound peer connection.
         tr("Inbound: initiated by peer"),
+        /*: Explanatory text for an Outbound peer connection which
```
s/Inbound/inbound/ and s/Outbound/outbound/ in these descriptions when used as an adjective and not as a proper noun, which seems to be the case throughout.

jonatack commented on src/qt/guiutil.cpp:
```diff
@@ -671,14 +671,23 @@ QString ConnectionTypeToQString(ConnectionType conn_type, bool prepend_direction
+        prefix = (conn_type == ConnectionType::INBOUND) ?
+                     //: An Inbound Connection from a Peer.
+                     QObject::tr("Inbound") :
+                     //: An Outbound Connection to a Peer.
```
here and above, s/Peer/peer/

jonatack commented on src/qt/rpcconsole.cpp:
```diff
@@ -471,14 +471,28 @@ RPCConsole::RPCConsole(interfaces::Node& node, const PlatformStyle *_platformSty
             .arg("addnode")
             .arg(QString(nonbreaking_hyphen) + "addnode")
             .arg(QString(nonbreaking_hyphen) + "connect"),
+        /*: Explanatory text for a temporary Outbound peer connection which
```
throughout: either s/ which/, which/ or s/which/that/

jonatack commented on src/qt/guiutil.cpp:
```diff
@@ -671,14 +671,23 @@ QString ConnectionTypeToQString(ConnectionType conn_type, bool prepend_direction
     case ConnectionType::BLOCK_RELAY: return prefix + QObject::tr("Block Relay");
+    //: Peer connection type established manually through one of several methods.
     case ConnectionType::MANUAL: return prefix + QObject::tr("Manual");
+    //: Peer connection type which feels or tests known addresses for validity.
```
would remove "feels" and maybe use "verifies", as "tests" can be either a noun or a verb and so is more ambiguous

jonatack commented on src/qt/guiutil.cpp:
```diff
@@ -671,14 +671,23 @@ QString ConnectionTypeToQString(ConnectionType conn_type, bool prepend_direction
     case ConnectionType::INBOUND: return prefix;
+    //: Peer connection type which relays all network information.
     case ConnectionType::OUTBOUND_FULL_RELAY: return prefix + QObject::tr("Full Relay");
+    //: Peer connection type which only relays Block network information.
```
throughout: s/Block/block/ as it is not a proper noun, unless part of one like "Outbound Block Relay", e.g. a connection type

jonatack reviewed (commented):
Thanks @jarolrod for reminding me to look at this.

ShaMan239 reviewed (commented):
Concept ACK

I like the changes that are done here. They would very much help a translator to translate the strings with proper context correctly.
However, I want to make some suggestions here:
- In the Guiutil.cpp file in lines 675 to 677: It would help a translator if the meaning of inbound and outbound was briefly explained there.
- In the rpconsole.cpp file at line number 494: It should be mentioned that the Outbound peer connection is temporary, as mentioned in line 491

jarolrod commented:
Updated from  f5c2377 ->  87b8129 ([pr345.02](https://github.com/jarolrod/gui/tree/pr345.02) -> [pr345.03](https://github.com/jarolrod/gui/tree/pr345.03))

Thanks @hebasto @jonatack @ShaMan239 for the reviews. I have implemented all of the review suggestions.

jarolrod commented:
updated from 87b8129 ->  07adcbe ([pr345.03](https://github.com/jarolrod/gui/tree/pr345.03) -> [pr345.04](https://github.com/jarolrod/gui/tree/pr345.04))

Changes: small cleanup with capitalized `Inbound Connection`

Talkless reviewed (approved):
Code review ACK, comments looks good.

hebasto commented:
@Talkless 
> Code review ACK, comments looks good.

Please insert a top commit hash :)

hebasto commented on src/qt/guiutil.cpp:
```diff
@@ -671,14 +671,26 @@ QString ConnectionTypeToQString(ConnectionType conn_type, bool prepend_direction
-        prefix = (conn_type == ConnectionType::INBOUND) ? QObject::tr("Inbound") : QObject::tr("Outbound") + " ";
+        prefix = (conn_type == ConnectionType::INBOUND) ?
+                     /*: An inbound connection from a peer. An inbound connection
+                         is a connection initated by a peer. */
```
typo: initated ==> initiated

Talkless replied:
Coll, thanks!

jarolrod replied:
Thanks updated

jonatack commented on src/qt/rpcconsole.cpp:
```diff
@@ -471,14 +471,28 @@ RPCConsole::RPCConsole(interfaces::Node& node, const PlatformStyle *_platformSty
+            relays all network information. This is the default behavior for
+            Outbound connections. */
         tr("Outbound Full Relay: default"),
+        /*: Explanatory text for an outbound peer connection which relays
```
throughout: either precede "which" with a comma and space, or use "that", e.g.

"connection, which relays"

"connection that relays"

jarolrod replied:
thanks, updated

jonatack commented:
(Maybe update the translations in merged #317 per the writing style here, if relevant.)

jarolrod commented:
updated from 07adcbe -> 8c13340 ([pr345.04](https://github.com/jarolrod/gui/tree/pr345.04) - > [pr345.05](https://github.com/jarolrod/gui/tree/pr345.05))

Changes: address @hebasto and @jonatack comments, thanks for the review!

@jonatack 
> (Maybe update the translations in merged #317 per the writing style here, if relevant.)

I will leave for follow-ups that improve translator comments throughout the GUI

shaavan commented on src/qt/guiutil.cpp:
```diff
@@ -671,14 +671,26 @@ QString ConnectionTypeToQString(ConnectionType conn_type, bool prepend_direction
     case ConnectionType::INBOUND: return prefix;
+    //: Peer connection type that relays all network information.
     case ConnectionType::OUTBOUND_FULL_RELAY: return prefix + QObject::tr("Full Relay");
+    /*: Peer connection type that relays relays network information about
```
The updated translated comment looks good. But I think there is some scope for correction. This might be an intentional decision by OP. But I still better mention it.

```suggestion
    /*: Peer connection type that relays relay network information about
```
or

```suggestion
    /*: Peer connection type that relays network information about
```

jarolrod replied:
was not intentional :D, thanks for catching this. Will update

jarolrod replied:
thanks, addressed in 371e2b9

jarolrod commented:
Updated from 8c13340 -> 371e2b9 ([pr345.05](https://github.com/jarolrod/gui/tree/pr345.05) -> [pr345.06](https://github.com/jarolrod/gui/tree/pr345.06), [diff](https://github.com/jarolrod/gui/compare/pr345.05..pr345.06))

Addressed @shaavan comments and removed a redundant word; thanks for the review!

shaavan reviewed (approved):
ACK 371e2b901bcbb1ab39f14431060f412030813757

The translator's comments are looking great! Kudos to @jarolrod!

jonatack commented on src/qt/rpcconsole.cpp:
```diff
@@ -471,14 +471,28 @@ RPCConsole::RPCConsole(interfaces::Node& node, const PlatformStyle *_platformSty
         tr("Inbound: initiated by peer"),
+        /*: Explanatory text for an outbound peer connection that
+            relays all network information. This is the default behavior for
+            Outbound connections. */
```
```suggestion
            outbound connections. */
```

jonatack commented on src/qt/guiutil.cpp:
```diff
@@ -671,14 +671,26 @@ QString ConnectionTypeToQString(ConnectionType conn_type, bool prepend_direction
     case ConnectionType::BLOCK_RELAY: return prefix + QObject::tr("Block Relay");
+    //: Peer connection type established manually through one of several methods.
     case ConnectionType::MANUAL: return prefix + QObject::tr("Manual");
+    //: Peer connection type that verifies or tests known addresses for validity.
```
would choose between "verifies" and "tests". also, see the definition in net.h#L160
```suggestion
    //: Short-lived peer connection type that tests the aliveness of known addresses.
```

jonatack commented on src/qt/guiutil.cpp:
```diff
@@ -671,14 +671,26 @@ QString ConnectionTypeToQString(ConnectionType conn_type, bool prepend_direction
     case ConnectionType::MANUAL: return prefix + QObject::tr("Manual");
+    //: Peer connection type that verifies or tests known addresses for validity.
     case ConnectionType::FEELER: return prefix + QObject::tr("Feeler");
+    //: Peer connection type that solicits known addresses from a peer.
```
```suggestion
    //: Short-lived peer connection type that solicits known addresses from a peer.
```

jonatack commented on src/qt/rpcconsole.cpp:
```diff
@@ -471,14 +471,28 @@ RPCConsole::RPCConsole(interfaces::Node& node, const PlatformStyle *_platformSty
             .arg("addnode")
             .arg(QString(nonbreaking_hyphen) + "addnode")
             .arg(QString(nonbreaking_hyphen) + "connect"),
+        /*: Explanatory text for a temporary outbound peer connection that
```
```suggestion
        /*: Explanatory text for a short-lived outbound peer connection that
```

jonatack commented on src/qt/rpcconsole.cpp:
```diff
@@ -471,14 +471,28 @@ RPCConsole::RPCConsole(interfaces::Node& node, const PlatformStyle *_platformSty
             .arg(QString(nonbreaking_hyphen) + "addnode")
             .arg(QString(nonbreaking_hyphen) + "connect"),
+        /*: Explanatory text for a temporary outbound peer connection that
+            is used to test the validity of known addresses. */
```
"temporary" seems a bit vague; could be true for most connections
```suggestion
            is used to test the aliveness of known addresses. */
```

jarolrod replied:
I like this word here, addressed in 4832737

jonatack commented on src/qt/rpcconsole.cpp:
```diff
@@ -471,14 +471,28 @@ RPCConsole::RPCConsole(interfaces::Node& node, const PlatformStyle *_platformSty
+        /*: Explanatory text for a temporary outbound peer connection that
+            is used to test the validity of known addresses. */
         tr("Outbound Feeler: short-lived, for testing addresses"),
+        /*: Explanatory text for a temporary outbound peer connection that is used
```
```suggestion
        /*: Explanatory text for a short-lived outbound peer connection that is used
```

jonatack reviewed (commented):
ACK 371e2b901bcbb

modulo some comments below, thanks for improving this!

jarolrod commented:
updated from 371e2b9 -> 4832737 ([pr345.06](https://github.com/jarolrod/gui/tree/pr345.06) -> [pr345.07](https://github.com/jarolrod/gui/tree/pr345.07), [diff](https://github.com/jarolrod/gui/compare/pr345.06..pr345.07)

jonatack commented:
Code review re-ACK 4832737c7dcc87afea5e1e88945ec311417aa876 per `git diff 371e2b9 4832737`, changes are translator comment edits since my review yesterday (thank you for updating)

hebasto reviewed (approved):
ACK 4832737c7dcc87afea5e1e88945ec311417aa876
